src/v3.rs

```diff
@@ -20,15 +20,14 @@ impl Uuid {
     /// [`NAMESPACE_URL`]: #associatedconstant.NAMESPACE_URL
     /// [`NAMESPACE_X500`]: #associatedconstant.NAMESPACE_X500
     pub fn new_v3(namespace: &Uuid, name: &[u8]) -> Uuid {
-        let mut context = md5::Context::new();
+        let mut hasher = md5::Md5::new();
 
-        context.consume(namespace.as_bytes());
-        context.consume(name);
+        hasher.update(namespace.as_bytes());
+        hasher.update(name);
 
-        let computed = context.compute();
-        let bytes = computed.into();
+        let computed = hasher.finalize();
 
-        let mut builder = crate::Builder::from_bytes(bytes);
+        let mut builder = crate::Builder::from_slice(&computed[..16]).unwrap();
```

Comment:
IIUC you don't need the slicing here, i.e. you can replace `&computed[..16]` with `&computed`. Same goes for SHA-1.